plugins/catalog-react/src/filters.ts

```diff
@@ -200,3 +201,25 @@ export class EntityErrorFilter implements EntityFilter {
     return error !== undefined && this.value === error;
   }
 }
+
+/**
+ * Filters entities based on the metadata category (status, domain, capabilities etc) provided.
+ * @public
+ */
+export class EntityFieldFilter implements EntityFilter {
+  constructor(readonly values: string[], readonly filterValue: string) {}
+  filterEntity(entity: Entity): boolean {
+    return this.values.every(v => {
+      let path = this.filterValue.split('.');
+      let val = '';
+      if (this.filterValue.startsWith('metadata.annotations.')) {
+        val = this.filterValue.substring('metadata.annotations.'.length);
+        path = ['metadata', 'annotations', val];
+      } else if (this.filterValue.startsWith('metadata.labels.')) {
+        val = this.filterValue.substring('metadata.labels.'.length);
+        path = ['metadata', 'labels', val];
+      }
```

Comment:
I had to do this because the keys in the metadata.annotations has `.` in it, which makes the lodash filter to fail. For example:
`"annotations": {
                "backstage.io/managed-by-location": "ams:https://ams-metadata-ql.prod.nimbus.expedia.com/graphql",
            }`

This applies for `metadata.labels` too.

Reply:
I was thinking we'd make the `filterValue` an array itself, that way we should be able to avoid this workaround